File: src/ProjectR/content/scripts/spells/CleansingSong.lua
function GetTargetType()
  return Single
end

function GetName()
  return "Cleansing Song"
end

function GetDescription()
  return "Sing a nice melody that removes all status debuffs of the target\nand converts stat debuffs into buffs."
end

function IsSupportSpell()
    return true
end

function GetMPCost()
    return .4
end

function GetDelay()
  return .65
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:RemoveDebuffs()
  for i = HP, SIL do
    local battleMod = ds:GetSingleStat(i):Get(BattleMod)
    if battleMod < 1 then
      d:BuffStat(i, 1 - battleMod)
    end
  end
end

File: src/ProjectR.Scripting/content/scripts/spells/CleansingSong.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class CleansingSong : SpellScriptBase
    {
        public override string Name { get { return "Cleansing Song"; } }
        public override string Description { get { return "Sing a nice melody that removes all status debuffs of the target\nand converts stat debuffs into buffs."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 40; } }
        public override double Delay { get { return .65; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            RemoveDebuffs();
            for (var stat = Stat.HP; stat < Stat.Count; ++ stat)
            {
                var battleMod = target.Stats[stat][StatType.BattleMod];
                if (battleMod < 1)
                {
                    BuffStat(stat, 1 - battleMod);
                }
            }
        }
    }
}